src/plugins/intel_gpu/src/graph/graph_optimizer/mark_runtime_skippable_nodes.cpp

```diff
@@ -18,6 +18,9 @@ void mark_runtime_skippable_nodes::run(program& p) {
         // Set gathers that might be skipped at runtime as can_be_optimized.
         // If not set, memory dependency will not work for the nodes that are skipped at runtime
         program_helpers::do_for_types<gather>(*node, [](gather_node& node){
+            // Skip if gather is impl_types::cpu
+            if (node.get_preferred_impl_type() == impl_types::cpu)
+                return;
```

Comment:
the original issue was not only for gather. Please apply the skipping in line 18